Add support for 'submit' input type in the crawler

anujanegi opened the pull request:
Fixes #65.

marco-c commented:
Can you upload screenshots for a page where there is a submit input? Upload the screenshot before pressing on the submit input and after.
If you can't find such a page, you can write a simple example yourself and make the crawler run on that example.

anujanegi commented:
Can you please guide me as to running the crawler on say "test.html".

anujanegi commented:
@marco-c  Hey, i tried but failed to do. I am not really able to find the problem. No screen shots are captured at all
.

marco-c commented:
Do you see any error? Maybe commit the code you have so I can see it (and then you will remove it later).

anujanegi commented:
@marco-c The error is -
Traceback (most recent call last):
  File "train.py", line 16, in <module>
    image = utils.load_image(all_images[0])
IndexError: list index out of range
Also, I committed the change for testing it. Can you please check.

marco-c commented:
>  File "train.py", line 16, in 

You don't need to run train.py in order to run the crawler, you just need to run collect.py.

marco-c commented on collect.py:
```diff
@@ -197,7 +199,7 @@ def do_something(driver, elem_properties=None):
     wait_loaded(driver)
-
+    file_path = "file:///home/anuja/tj/autowebcompat/test.html"
     driver.get_screenshot_as_file(file_path)
```
You are changing where the screenshot must be saved, you need to change **driver.get** to load the URL (a few lines after).

anujanegi commented:
With no changes in collect.py, when I run it - 
     Using TensorFlow backend.
     10280
I guess that the script is only running till a certain point, where it prints the length of bugs using utils.get_bugs().

marco-c commented:
You do need to make changes to collect.py if you want it to collect the screenshot for your test page.
The main change is to **driver.get**.

anujanegi commented:
I updated collect.py as required. Can you review.
It should ideally work this way I think. But doesn't, stops exec after a point.
    Using TensorFlow backend.
    10280

marco-c commented on collect.py:
```diff
@@ -308,4 +307,4 @@ def main(bugs):
     random.shuffle(bugs)
     with ThreadPoolExecutor(max_workers=MAX_THREADS) as executor:
         for i in range(MAX_THREADS):
-            executor.submit(main, bugs[i::MAX_THREADS])
```
I think the resulting code here is not valid Python, it's strange that it even runs.

Just remove the entire threadpoolexecutor block and call main(bugs).

anujanegi commented:
Hey, i thought so too and tried doing it likewise. Ran into this -
Using TensorFlow backend.
10280
Traceback (most recent call last):
  File "collect.py", line 307, in <module>
    main(bugs)
  File "collect.py", line 301, in main
    firefox_driver = webdriver.Firefox(firefox_profile=firefox_profile, firefox_binary=nightly_bin)
  File "/usr/local/lib/python3.5/dist-packages/selenium/webdriver/firefox/webdriver.py", line 80, in __init__
    self.binary, timeout)
  File "/usr/local/lib/python3.5/dist-packages/selenium/webdriver/firefox/extension_connection.py", line 52, in __init__
    self.binary.launch_browser(self.profile, timeout=timeout)
AttributeError: 'str' object has no attribute 'launch_browser'
Tried searching too, but had no luck.

marco-c commented:
Are you using the versions of the packages that are specified in requirements.txt?

anujanegi commented:
Selenium was not the correct version, my bad. But now the error thrown is - 
/usr/local/lib/python3.5/dist-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Using TensorFlow backend.
10280
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/selenium/webdriver/common/service.py", line 74, in start
    stdout=self.log_file, stderr=self.log_file)
  File "/usr/lib/python3.5/subprocess.py", line 947, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.5/subprocess.py", line 1551, in _execute_child
    raise child_exception_type(errno_num, err_msg)
FileNotFoundError: [Errno 2] No such file or directory: 'geckodriver'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "collect.py", line 307, in <module>
    main(bugs)
  File "collect.py", line 301, in main
    firefox_driver = webdriver.Firefox(firefox_profile=firefox_profile, firefox_binary=nightly_bin)
  File "/usr/local/lib/python3.5/dist-packages/selenium/webdriver/firefox/webdriver.py", line 142, in __init__
    self.service.start()
  File "/usr/local/lib/python3.5/dist-packages/selenium/webdriver/common/service.py", line 81, in start
    os.path.basename(self.path), self.start_error_message)
selenium.common.exceptions.WebDriverException: Message: 'geckodriver' executable needs to be in PATH.

marco-c commented:
Did you setup the project as explained in the README? It looks like you don't have geckodriver, which should be there if you clone the project properly.

anujanegi commented:
Caught hold and fixed all my stupid mistakes. Hosted a test html file to verify changes. Attaching proofs below. Also, removed all the modifications that had been made for testing and pushed the final fix.
![image](https://user-images.githubusercontent.com/29161453/38045496-8de8153c-32da-11e8-9441-39d97fe6b40c.png)
![image](https://user-images.githubusercontent.com/29161453/38045618-e40da8f0-32da-11e8-9711-1f9263758b7b.png)

![image](https://user-images.githubusercontent.com/29161453/38045540-b1f79736-32da-11e8-905e-2d64d21bb2fc.png)
